### PR TITLE
Functional Enhancements (v2)

### PR DESCRIPTION
**Pull request by RandallSteck**
Enhance vector processing -- 
1) Modify main vector processor to extract vsId from filename if not found in file.  Proxy-style files do not have vsId in the file, and libacvp requires vsId for processing.  If a vsId does not exist in the file, the filename will be searched for "vs" followed by digits, which will be interpreted as a vsId.  This was an early patch for proxy-style files that works well enough.

2) New function to process vectors that specifically supports acvpproxy-style vector files.  It is a 1st attempt at a "universal" vector file processor.  The function is named "offline" because it ignores the header (jwt, vsid, etc) and gets right to processing whatever it finds. It will process an array of algorithms (libacvp-style) or a single algorithm (acvpproxy style).  It is resilient to the occasional degenerate case of a naked (non-arrayed) json object. It is called "offline" since we're not tracking any of the session data.  The function is probably not usable (and indeed isn't required) in the case of a single-command full-round-trip testing session. That said, I have not blocked it off with ACVP_OFFLINE.  

Add domain-type for PT length in symmetric ciphers -- Protocol allows domains, libacvp allowed only value arrays.

Add ACVP "health check" function -- An easy way to verify stable communications to an ACVP server without invoking any vector processing

Minor other tweaks and typos

### CODE REVIEW

File: src/acvp.c [hunk 1]
```diff
@@ -1089,6 +1094,20 @@ ACVP_RESULT acvp_run_vectors_from_file(ACVP_CTX *ctx, const char *req_filename,
         if (rv != ACVP_SUCCESS) goto end;
         ACVP_LOG_INFO("Received vsid_url=%s", vsid_url);
     }
+    /* If no vsIds found, try to get it from the filename */
+    if (vs_cnt == 0) {
+        int vsid;
+        char* start = strstr(req_filename, "vs");
+        if (!start || (0 == (vsid = atoi(start+2)))) {
```
Comment (abkarcher) [inline src/acvp.c:1101]: could we add a check to make sure start+2 does not go past end of string?

File: src/acvp_util.c [hunk 2]
```diff
@@ -1563,10 +1563,12 @@ ACVP_RESULT acvp_json_serialize_to_file_pretty_a(const JSON_Value *value, const
             fclose(fp);
             return ACVP_JSON_ERR;
         }
+        if (!first) {  // Do not add "," before this data
```
Comment (abkarcher) [inline src/acvp_util.c:1566]: can we combine these nested if statements? static analysis will complain :)

**Comment by abkarcher** [comment]
the "universal" vector set parsing works fantastically, thanks! After we sort out the other comments, can you re-open this PR against 2_2_0-throttle? Typically, we want to keep API additions and changes to new minor release versions when possible.

Thanks,
Andrew

**Comment by RandallSteck** [comment]
Closed to be re-open against 2_2_0

**Pull request closed without merging**